code/modules/reagents/reagent_containers/hypospray.dm

```diff
@@ -28,6 +28,10 @@
 		return FALSE
 	if(!iscarbon(M))
 		return FALSE
+	//Checks if the survival pen from mining is used in a low pressure environment.
+	if((!lavaland_equipment_pressure_check(get_turf(user))) && istype(src, /obj/item/reagent_containers/hypospray/medipen/survival))

```

Comment:
Do a var or flag check instead of an istype one.
For example, add something like `var/hypo_flags = NONE` to the base of it, and then add `hypo_flags = LAVALAND_RESTRICTED` in `/obj/item/reagent_containers/hypospray/medipen/survival`
The check then becomes something like this: `if(hypo_flags & LAVALAND_RESTRICTED && !lavaland_equipment_pressure_check(get_turf(user)))`

Reply:
On it !

Reply:
Done, will merge.